replicatedhq/kURL
# Migrate Internal Snapshots from Object Store to PVC

If you install with Velero 1.6.2 from this branch, the new default will be to use the PVC-backed snapshots.

To Test Migration:
1. Install Rook Spec with flag enabled: `curl https://kurl.sh/4b91c41 | sudo bash`
2. Take snapshots through the kotsadm console
3. Run the upgrade locally using this manifest and the install command: `cat install.sh | sudo BETA_VELERO_USE_INTERNAL_PVC=true REGISTRY_USE_PVC_STORAGE=1 bash`
```
INSTALLER_YAML="apiVersion: cluster.kurl.sh/v1beta1
kind: Installer
metadata: 
  name: test
spec: 
  kubernetes: 
    version: 1.20.11
  weave: 
    version: 2.6.5
  containerd: 
    version: 1.4.6
  longhorn: 
    version: 1.1.2
  registry:
    version: 2.7.1
  velero:
    version: 1.6.2
  kotsadm:
    version: 1.57.0
    disableS3: true"
```
4. Go the kotsadm console and verify the snapshots are still there
5. Perform a full restore and observe the app and console are restored.

Additional PR coming to backport to other Velero versions and automated template.
